src/averageAges.js

```diff
@@ -34,7 +38,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people
+    .map((person) => person.mother)
+    .filter((name) => !!name);
+
+  const women = people
+    .filter((person) => withChildren
+      ? person.sex === 'f' && mothers.includes(person.name)
+      : person.sex === 'f');
```

Comment:
```suggestion
  const women = people.filter(person =>
    withChildren
      ? people.some(child => child.mother === person.name) && person.sex === 'f'
      : person.sex === 'f'
  );
```
You could use the method `.some` inside your method `.filter` and avoid creating another variable 'mother'. It would make your code more simple and shorter